kratos/utilities/delaunator_utilities.cpp

```diff
@@ -234,6 +234,77 @@ std::vector<std::size_t> ComputeTrianglesConnectivity(const std::vector<Point>&
     return connectivities;
 }
 
+/***********************************************************************************/
+/***********************************************************************************/
+
+std::pair<std::vector<std::size_t>, std::vector<double>> ComputeTrianglesConnectivity(
+    const std::vector<double>& rCoordinates,
+    const std::vector<std::array<double,2>>& rSegments,
+    const double AreaConstraint)
+{
+    // Creating the containers for the input and output
+    struct triangulateio in_mid, out_mid, vorout_mid;
+
+    InitializeTriangulateIO(in_mid);
+    InitializeTriangulateIO(out_mid);
+    InitializeTriangulateIO(vorout_mid);
+
+    // Initialize the boundary points coordinates list
+    in_mid.numberofpoints = rCoordinates.size()/2;
+    in_mid.pointlist = (REAL *) malloc(in_mid.numberofpoints * 2 * sizeof(REAL));
+
+    for(std::size_t i = 0; i < rCoordinates.size(); ++i) {
+        in_mid.pointlist[i] = rCoordinates[i];
+    }
+
+    // Initilize the segment list (note that default zero markers are assumed)
+    in_mid.numberofsegments = rSegments.size();
+    in_mid.segmentlist = (int*) malloc(in_mid.numberofsegments * 2 * sizeof(int));
+    in_mid.segmentmarkerlist = (int*) malloc(in_mid.numberofsegments * sizeof(int));
+    for (std::size_t i = 0; i < in_mid.numberofsegments; ++i) {
+        const auto& r_segment = rSegments[i];
+        in_mid.segmentlist[2*i] = r_segment[0];
+        in_mid.segmentlist[2*i + 1] = r_segment[1];
+    }
+
+    // Check https://www.cs.cmu.edu/~quake/triangle.switch.html for a detailed Triangle switches description
+    // "Q" quiet (no terminal output except errors)
+    // "q" quality mesh generation with no angles smaller than 20 degrees
+    // "p" triangulates a Planar Straight Line Graph
+    // "z" numbers all items starting from zero (rather than one)
+    // "a" imposes a maximum triangle area constrain
+    std::string str = AreaConstraint > 0.0 ? "Qqpza" + std::to_string(AreaConstraint) : "Qqpz";
+    char *meshing_options = new char[str.length() + 1];
+    strcpy(meshing_options, str.c_str());
+    triangulate(meshing_options, &in_mid, &out_mid, &vorout_mid);
```

Comment:
```suggestion
    const std::string meshing_options = AreaConstraint > 0.0 ? "Qqpza" + std::to_string(AreaConstraint) : "Qqpz";
    triangulate(meshing_options.c_str(), &in_mid, &out_mid, &vorout_mid);
```

Reply:
This does not work, unfortunately. The used library triangle expects a C-style char and not strings.

Reply:
I am pretty sure it does ;)
Check what the [`c_str](https://en.cppreference.com/w/cpp/string/basic_string/c_str) does:
it returns the `char*`

Reply:
it returns `const char*` and not `char*`, which in this case gives errors. 
See the link below: 
https://www.geeksforgeeks.org/convert-string-char-array-cpp/

Reply:
ah ok

well then I would do it as the link you show without the dynamic memory

does triangle modify this string? or did they just forget the `const` in their interface?

Reply:
or even the third method in your link:
`&str_obj[0]`

Reply:
triangle does not touch the string. I'll change it according to the third link. Thanks Philipp